classes/utils.php

```diff
@@ -622,13 +582,13 @@ public static function update_studentquiz_progress(\stdClass $studentquizprogres
      * @param int $timecreated The time do action.
      * @return bool|int True or new id
      */
-    public static function question_save_action(int $studentquizquestionid, int $userid = null, int $state,
+    public static function question_save_action(int $studentquizquestionid, ?int $userid, int $state,
             int $timecreated = null) {
-        global $DB, $USER;
+        global $DB;
 
         $data = new \stdClass();
         $data->studentquizquestionid = $studentquizquestionid;
-        $data->userid = isset($userid) ? $userid : $USER->id;
+        $data->userid = $userid;
```

Comment:
Is this change really safe? The funciton still seems to allow userid = null to be passed in, so don't we need this?

Reply:
Oh, I see. The DB scheme was changed to allow null. OK.